pkg/database/db_seed.go

```diff
@@ -148,228 +192,518 @@ func seedDefaultTags() map[string]models.Tag {
+
+	// Process tags for this quickstart
+	return processQuickstartTags(template, quickstart, result)
 }
 
-func SeedTags() {
-	// clear old content pahse
-	favorites := clearOldContent()
-	// seeding phase
-	defaultTags := seedDefaultTags()
-	MetadataTemplates := findTags()
-
-	for _, template := range MetadataTemplates {
-		kind := template.Kind
-		if kind == "QuickStarts" {
-			var quickstart models.Quickstart
-			var quickstartErr error
-			var tags []models.Tag
-			quickstart, quickstartErr = seedQuickstart(template, defaultTags["quickstart"])
-			if quickstartErr != nil {
-				fmt.Println("Unable to seed quickstart: ", quickstartErr.Error(), template.ContentPath)
-			}
-			// Clear all tags associations
-			quickstart.Tags = tags
-			DB.Save(&quickstart)
-
-			for _, tag := range template.Tags {
-				var newTag models.Tag
-				var originalTag models.Tag
-				newTag.Type = models.TagType(tag.Kind)
-				newTag.Value = tag.Value
-
-				r := DB.Preload("Quickstarts").Where("type = ? AND value = ?", models.TagType(newTag.Type), newTag.Value).Find(&originalTag)
-				if r.Error != nil {
-					fmt.Println("Error: ", r.Error.Error())
-				} else if r.RowsAffected == 0 {
-					DB.Create(&newTag)
-					originalTag = newTag
-				}
+// processQuickstartTags handles tag processing for a quickstart
+func processQuickstartTags(template MetadataTemplate, quickstart models.Quickstart, result *SeedingResult) error {
+	for _, tag := range template.Tags {
+		var newTag models.Tag
+		var originalTag models.Tag
+		newTag.Type = models.TagType(tag.Kind)
+		newTag.Value = tag.Value
 
-				// Create tags quickstarts associations
-				err := DB.Model(&originalTag).Association("Quickstarts").Append(&quickstart)
-				if err != nil {
-					fmt.Println("Failed creating tags associations", err.Error())
-				}
+		logrus.Debugf("Processing tag: %s=%s for quickstart %s", tag.Kind, tag.Value, quickstart.Name)
 
-				quickstart.Tags = append(quickstart.Tags, originalTag)
+		r := DB.Preload("Quickstarts").Where("type = ? AND value = ?", models.TagType(newTag.Type), newTag.Value).Find(&originalTag)
+		if r.Error != nil {
+			errMsg := fmt.Sprintf("failed to lookup tag %s=%s: %v", tag.Kind, tag.Value, r.Error)
+			logrus.Error(errMsg)
+			result.addError(errMsg)
+			continue
+		}
 
-				DB.Save(&quickstart)
-				DB.Save(&originalTag)
+		if r.RowsAffected == 0 {
+			// Create new tag
+			if err := DB.Create(&newTag).Error; err != nil {
+				errMsg := fmt.Sprintf("failed to create tag %s=%s: %v", tag.Kind, tag.Value, err)
+				logrus.Error(errMsg)
+				result.addError(errMsg)
+				continue
 			}
+			originalTag = newTag
+			result.TagsCreated++
+			logrus.Debugf("Created new tag: %s=%s", tag.Kind, tag.Value)
+		}
+
+		// Create tag-quickstart association
+		if err := DB.Model(&originalTag).Association("Quickstarts").Append(&quickstart); err != nil {
+			errMsg := fmt.Sprintf("failed to create tag association %s=%s for quickstart %s: %v", tag.Kind, tag.Value, quickstart.Name, err)
```

Comment:
Since we didn't have error handling prior it's hard to tell what the flow prior to these changes were (returning early or not). As long things are working by the end this should be fine, but something to think about.